src/schematools/contrib/django/models.py

```diff
@@ -224,14 +226,26 @@ def __init__(self, *args, **kwargs):
             self.schema_data if "schema_data" in self.__dict__ and not self._state.adding else None
         )
 
+    @classmethod
+    def name_from_schema(cls, schema: DatasetSchema) -> str:
+        """Generate dataset name from schema"""
+        name = to_snake_case(schema.id)
+        if schema.version and not schema.is_default_version:
+            name = to_snake_case(f"{name}_{schema.version}")
```

Comment:
perhaps an `@` to concat name with version?

Reply:
Was planning to use it at first, but it will break wonderfully every time we use `to_snake_case`.

Reply:
ah I checked and missed it: `from schematools import MAX_TABLE_NAME_LENGTH, RELATION_INDICATOR, TMP_TABLE_POSTFIX, types`

there is a ugly hack that might work:

```
from __future__ import annotations
import typing

...

# at bottom of the file:
if typing.TYPE_CHECKING:
    from schematools import  types
```